keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -0,0 +1,2262 @@
+supported by the ACME GPU hardware. I know that the administrator has created
+an "acme-gpu" ResourceClass.
+
+For a simple trial, I create a Pod directly where two containers share the same subset
+of the GPU:
+```
+apiVersion: gpu.acme.com/v1
+kind: GPURequirements
+metadata:
+  name: device-consumer-gpu-parameters
+memory: "2Gi"
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: device-consumer
+spec:
+  resourceClaims:
+  - name: "gpu" # this name gets referenced below under "claims"
+    template:
+      resourceClassName: "acme-gpu"
+      parameters:
+        apiVersion: gpu.acme.com/v1
+        kind: GPURequirements
+        name: device-consumer-gpu-parameters
+  containers:
+  - name: workload
+    image: my-app
+    command: ["/bin/program"]
+    resources:
+      requests:
+        memory: "64Mi"
+        cpu: "250m"
+      limits:
+        memory: "128Mi"
+        cpu: "500m"
+      claims:
+        - "gpu"
+  - name: monitor
+    image: my-app
+    command: ["/bin/other-program"]
+    resources:
+      requests:
+        memory: "32Mi"
+        cpu: "25m"
+      limits:
+        memory: "64Mi"
+        cpu: "50m"
+      claims:
+      - "gpu"
+```
+
+This request triggers resource allocation on a node that has a GPU device with
+2Gi of memory available and then the Pod runs on that node. The remaining
+capacity of the GPU may be usable for other pods, with constrains like alignment
+to segment sizes ensured by the resource driver.
+The lifecycle of the resource
+allocation is tied to the lifecycle of the Pod.
+
+In production, a similar PodTemplateSpec in a Deployment will be used.
+
+#### Combined setup of different hardware functions
+
+As a 5G telco operator, I want to use the FPGA IP block, signal processor and
+network interfaces provided by the [Intel FPGA
+N3000](https://www.intel.com/content/www/us/en/products/details/fpga/platforms/pac/n3000.html)
+card in a Kubernetes edge cluster. Intel provides a single resource driver that
+has parameters for setting up all of these hardware functions together as
+needed for a data flow pipeline.
+
+### Notes/Constraints/Caveats
+
+Scheduling is likely to be slower when many Pods request the new
+resource types, both because scheduling such a Pod involves more
+round-trips through the API server for ResourceClaimStatus updates and
+because scheduling one Pod may affect other Pods in ways that cannot
+be anticipated by the kube-scheduler. When many Pods compete for
+limited resources, multiple attempts may be needed before a suitable
+node is found.
+
+The hardware that is expected to need this more flexible allocation
+approach is likely to be used by pods that run for extended periods of time,
+so this is not a major concern.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
```

Comment:
When will we answer this suggested question?

Reply:
I have reached out to SIG security on Slack this weekend - I know, I should have done that sooner. I've not heard back from anyone and don't know whom to ask directly.

Reply:
Ping Ian or Tabby and see if they have a few minutes for a first-impression?

Reply:
Which Ian and which Tabby? If you tag them here with their GitHub handle, then I can follow up on Slack.

Reply:
I pinged them privately on Slack to raise awareness for https://kubernetes.slack.com/archives/C019LFTGNQ3/p1655579249114199 (the #sig-security message).